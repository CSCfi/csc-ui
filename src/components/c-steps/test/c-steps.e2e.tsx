import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';

describe('c-steps', () => {
  let page: E2EPage;

  let steps: E2EElement;

  beforeEach(async () => {
    page = await newE2EPage({
      html: `
    <c-steps>
        <c-step>Step 1</c-step>
        <c-step>Step 2</c-step>
        <c-step>Step 3</c-step>
    </c-steps>`,
    });
    steps = await page.find('c-steps');
  });

  it('renders correctly', async () => {
    expect(steps).toHaveClass('hydrated');
  });

  it('renders completed steps', async () => {
    let completedSteps = await page.findAll('c-steps >>> c-step--complete');
    expect(completedSteps.length).toBe(0);

    // steps.setProperty('value', 3);
    await page.waitForChanges();

    completedSteps = await page.findAll('c-step');
    expect(completedSteps).toMatchSnapshot();
    expect(completedSteps.length).toBe(2);
  });
});
