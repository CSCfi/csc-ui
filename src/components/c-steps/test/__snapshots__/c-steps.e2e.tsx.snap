// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-steps renders completed steps 1`] = `
Array [
  <c-step class="hydrated">
  <mock:shadow-root>
    <div class="c-step c-step--complete">
      <div class="c-step__indicator">
        <div class="complete">
          <svg viewBox="0 0 100 100">
            <path class="path" d="M 12 52 l 24 24 l 47 -47 l -3 -3 l -44 44 l -21 -21 l -3 3"></path>
          </svg>
        </div>
      </div>
      <div class="c-step__label">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  Step 1
</c-step>,
  <c-step class="hydrated">
  <mock:shadow-root>
    <div class="c-step">
      <div class="c-step__indicator">
        <div class="dot"></div>
      </div>
      <div class="c-step__label">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  Step 2
</c-step>,
  <c-step class="hydrated">
  <mock:shadow-root>
    <div class="c-step">
      <div class="c-step__indicator">
        <div class="dot"></div>
      </div>
      <div class="c-step__label">
        <slot></slot>
      </div>
    </div>
  </mock:shadow-root>
  Step 3
</c-step>,
]
`;
